openfoodfacts/api.py

```diff
@@ -450,6 +450,21 @@ def __init__(
         self.facet = FacetResource(self.api_config)
         self.robotoff = RobotoffResource(self.api_config)
 
+    def search_advanced(self, query: str, filters: dict = None, page: int = 1):
+        """Advanced product search using Search-a-licious endpoint."""
+        filters = filters or {}
+        params = {"search_terms": query, "page": page, **filters}
```

Comment:
The search terms should be provided using the `q` parameter, see the search-a-licious documentation here: https://search.openfoodfacts.org/docs#/default/search_get_search_get